kasa/tests/fakeprotocol_smart.py

```diff
@@ -430,14 +430,24 @@ def _edit_preset_rules(self, info, params):
         info["get_preset_rules"]["states"][params["index"]] = params["state"]
         return {"error_code": 0}
 
+    def _update_sysinfo_key(self, info: dict, key: str, value: str) -> dict:
+        """Update a single key in the main system info.
+
+        This is used to implement child device setters that change the main sysinfo state.
+        """
+        sys_info = info.get("get_device_info", info)
+        sys_info[key] = value
+
+        return {"error_code": 0}
+
     async def _send_request(self, request_dict: dict):
         method = request_dict["method"]
 
         info = self.info
         if method == "control_child":
             return await self._handle_control_child(request_dict["params"])
 
-        params = request_dict.get("params")
+        params = request_dict.get("params", {})
```

Comment:
I think we can always assume that there's at least an empty dict as params? This way we can avoid checking for Noneness below.

Reply:
I think this is fine